packages/inference/src/lib/makeRequestOptions.ts

```diff
@@ -38,7 +45,45 @@ export function makeRequestOptions(
 		}
 	}
 
-	const url = /^http(s?):/.test(model) || model.startsWith("/") ? model : `${HF_INFERENCE_API_BASE_URL}${model}`;
+	const url = await (async () => {
+		// If model is an URL, do not change it
+		if (/^http(s?):/.test(model) || model.startsWith("/")) {
+			return model;
+		}
+
+		if (options?.task) {
+			const key = `${model}:${accessToken}`;
+			let cachedTask = taskCache.get(key);
+
+			if (cachedTask && cachedTask.date < new Date(Date.now() - CACHE_DURATION)) {
+				taskCache.delete(key);
+				cachedTask = undefined;
+			}
+
+			if (cachedTask === undefined) {
+				const task = await fetch(`${HF_HUB_URL}/api/models/${model}?expand[]=pipeline_tag`, {
+					headers: accessToken ? { Authorization: `Bearer ${accessToken}` } : {},
+				})
+					.then((resp) => resp.json())
+					.then((json) => json.pipeline_tag)
+					.catch(() => null);
+
+				cachedTask = { task, date: new Date() };
+				taskCache.set(key, { task, date: new Date() });
+
+				if (taskCache.size > MAX_CACHE_ITEMS) {
+					taskCache.delete(taskCache.keys().next().value);
+				}
+			}
+
+			if (cachedTask.task !== options.task) {
+				return `${HF_INFERENCE_API_BASE_URL}pipeline/${options.task}/${model}`;
+			}
+		}
+
+		return `${HF_INFERENCE_API_BASE_URL}models/${model}`;
+	})();
```

Comment:
hi @coyotte508 , trying to understand this, why do you need to cache the task?

Reply:
Oh I see, just to check the task tag , but if the user is passing the task, do you even need to check the hub ?

Reply:
Here the task is automatically passed from the `featureExtraction`, `textClassification` or `sentenceSimilarity` function, it's not passed by the user.

Maybe we should add an param to differentiate direct calls to `request` from calls made by the lib wrappers.